src/app/reporting/ReportingEngine.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements reporting engine for CHIP
+ *      Data Model profile.
+ *
+ */
+
+#include <app/InteractionModelEngine.h>
+#include <app/reporting/ReportingEngine.h>
+
+namespace chip {
+namespace app {
+namespace reporting {
+CHIP_ERROR ReportingEngine::Init()
+{
+    CHIP_ERROR err       = CHIP_NO_ERROR;
```

Comment:
`return CHIP_NO_ERROR` at the end. No need to declare a separate variable.

Reply:
yep, done, thanks